## FINAL PROJECT SUBMISSION

- GROUP 5

STUDENT NAME:

**- OLIVE MULOMA**

**- ABIGAIL MWENDWA**

**- HAWKINS MURITHI**

**- HARRY ATULAH**

STUDENT PACE: *FULL TIME/HYBRID*

SCHEDULE PROJECT REVIEW DATE/TIME: *29TH JULY 2024*

INSTRUCTOR NAME: **MARYANN MWIKALI**


# NEWSNET PUBLISHING COMPANY

## BUSINESS OVERVIEW

- In today's digital age, news consumption has skyrocketed, and with it, the challenge of efficiently managing and categorizing vast amounts of content. Automated text classification offers a solution by enabling efficient sorting of news articles into relevant categories such as Politics, Technology, Entertainment, and Business. This enhances user experience by ensuring that readers have quick access to content that interests them and assists in organizing and managing content for media companies.

## IMPORT REQUIRED LIBRARIES

In [2]:
import pandas as pd 
import seaborn as sns
import numpy as np
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.model_selection import train_test_split
from sklearn.model_selection import cross_val_score
from sklearn.metrics import roc_auc_score, roc_curve, auc
from itertools import cycle
from sklearn.preprocessing import label_binarize
from gensim.models import Word2Vec
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import classification_report
import matplotlib.pyplot as plt
from sklearn.pipeline import Pipeline
from sklearn.compose import ColumnTransformer
from sklearn.feature_extraction.text import TfidfTransformer
from tensorflow.keras.preprocessing.text import Tokenizer
from tensorflow.keras.preprocessing.sequence import pad_sequences
from sklearn.preprocessing import StandardScaler
from sklearn.naive_bayes import MultinomialNB
from sklearn.metrics import classification_report, confusion_matrix
import string
import pickle
import nltk
from wordcloud import WordCloud
from nltk.corpus import stopwords
from nltk.tokenize import word_tokenize
from nltk.stem import WordNetLemmatizer
from flask import Flask, request, jsonify, render_template_string
import os
import warnings
warnings.filterwarnings('ignore')

## DATA UNDERSTANDING

In [3]:
data_test = pd.read_csv("data/Data_Test.csv", encoding='latin1')

data_test.head(10)

STORY
0  2019 will see gadgets like gaming smartphones ...
1  It has also unleashed a wave of changes in the...
2  It can be confusing to pick the right smartpho...
3  The mobile application is integrated with a da...
4  We have rounded up some of the gadgets that sh...
5  "Imagine if every message you sent was kept wi...
6  Positioned along the four sides of the Asus RO...
7  In fact, when I applied to USC film school the...
8  As spotted by Android Police, Netflix is testi...
9  Her moves were immaculately choreographed as s...

In [4]:
data_test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2748 entries, 0 to 2747
Data columns (total 1 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   STORY   2748 non-null   object
dtypes: object(1)
memory usage: 21.6+ KB


#### The DataFrame `data_test` has:

- All entries in the "STORY" column are non-null and contain text data
- The DataFrame occupies 21.6 KB of memory

**- Description:**  The test dataset contains 2,748 rows and 1 column: STORY.


**- STORY:** Contains the text of news articles, similar to the training dataset but without associated category labels.


**-Usage:** This dataset is used to evaluate the performance of the trained model. The STORY column will be input to the model to predict the category labels, which can then be compared to actual labels (if available) or used for making predictions on new data.

In [5]:
data_test.shape

(2748, 1)

In [6]:
data_train = pd.read_csv("data/Data_Train.csv", encoding='latin1')

data_train.head(10)

STORY  SECTION
0  But the most painful was the huge reversal in ...        3
1  How formidable is the opposition alliance amon...        0
2  Most Asian currencies were trading lower today...        3
3  If you want to answer any question, click on ‘...        1
4  In global markets, gold prices edged up today ...        3
5  BEIJING: Chinese tech giant Huawei has announc...        1
6  Mumbai: India Inc's external commercial borrow...        3
7  On Wednesday, Federal Reserve Chairman Jerome ...        3
8  What more can you give to the audience? I have...        2
9  com, Arbaaz Khan spoke about getting back to D...        2

In [7]:
data_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7628 entries, 0 to 7627
Data columns (total 2 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   STORY    7628 non-null   object
 1   SECTION  7628 non-null   int64 
dtypes: int64(1), object(1)
memory usage: 119.3+ KB
